web/src/components/molecules/Integration/IntegrationSettingsModal/index.tsx

```diff
@@ -1,39 +1,62 @@
+import { useCallback } from "react";
+
 import Button from "@reearth-cms/components/atoms/Button";
 import Form from "@reearth-cms/components/atoms/Form";
 import Modal from "@reearth-cms/components/atoms/Modal";
 import Select from "@reearth-cms/components/atoms/Select";
+import { IntegrationMember, Role } from "@reearth-cms/components/molecules/Integration/types";
 import { useT } from "@reearth-cms/i18n";
 
 export type FormValues = {
-  name: string;
+  role: Role;
 };
 
 export type Props = {
+  selectedIntegrationMember?: IntegrationMember;
   open?: boolean;
   onClose?: () => void;
-  onSubmit?: () => Promise<void> | void;
+  onSubmit?: (role: Role) => Promise<void> | void;
 };
 
-const IntegrationSettingsModal: React.FC<Props> = ({ open, onClose, onSubmit }) => {
+const IntegrationSettingsModal: React.FC<Props> = ({
+  open,
+  onClose,
+  onSubmit,
+  selectedIntegrationMember,
+}) => {
   const t = useT();
   const { Option } = Select;
   const [form] = Form.useForm();
 
+  const handleSubmit = useCallback(async () => {
+    try {
+      const values = await form.validateFields();
+      await onSubmit?.(values.role);
+      onClose?.();
+      form.resetFields();
+    } catch (info) {
+      console.log("Validate Failed:", info);
+    }
+  }, [form, onClose, onSubmit]);
+
   return (
     <Modal
-      title={t("Integration Setting")}
+      title={t("Integration Setting") + "  " + selectedIntegrationMember?.integration?.name}
       visible={open}
       onCancel={() => onClose?.()}
-      onOk={onSubmit}
+      onOk={handleSubmit}
```

Comment:
It doesn't seem like you are making use of the confirm functionality inside antd's modal, are you?
If not, the onOk isn't necessary since you are doing handleSubmit on the Save button click.

Reply:
I'm not using it in this modal, ok :+1: